cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go

```diff
@@ -233,7 +238,61 @@ func (mig *Mig) Nodes() ([]string, error) {
 
 // TemplateNodeInfo returns a node template for this node group.
 func (mig *Mig) TemplateNodeInfo() (*schedulercache.NodeInfo, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	template, err := mig.gceManager.getMigTemplate(mig)
+	if err != nil {
+		return nil, err
+	}
+	node, err := mig.gceManager.buildNodeFromTemplate(mig, template)
+	if err != nil {
+		return nil, err
+	}
+	nodeInfo := schedulercache.NewNodeInfo(buildKubeProxy(mig))
+	nodeInfo.SetNode(node)
+	return nodeInfo, nil
+}
+
+// Builds KubeProxy pod definition
+func buildKubeProxy(mig *Mig) *apiv1.Pod {
```

Comment:
Is this gce specific? If kubeProxy is similar everywhere maybe we should put it in some utils file?